src/dialogs/create-file.tsx

```diff
@@ -0,0 +1,232 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+
+import { Alert, AlertVariant } from '@patternfly/react-core/dist/esm/components/Alert';
+import { Button } from '@patternfly/react-core/dist/esm/components/Button';
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Modal, ModalVariant } from '@patternfly/react-core/dist/esm/components/Modal';
+import { TextArea } from '@patternfly/react-core/dist/esm/components/TextArea';
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { Stack } from '@patternfly/react-core/dist/esm/layouts/Stack';
+
+import cockpit from 'cockpit';
+import { FormHelper } from 'cockpit-components-form-helper';
+import type { Dialogs, DialogResult } from 'dialogs';
+import { useInit } from 'hooks';
+import { superuser } from 'superuser';
+
+import "./editor.scss";
+import { useFilesContext } from '../app.tsx';
+import { checkFilename } from '../common.ts';
+import { get_owner_candidates } from '../ownership.tsx';
+
+import { edit_file } from './editor.tsx';
+
+const _ = cockpit.gettext;
+
+async function create_file(filename: string, content?: string, owner?: string) {
+    const file = cockpit.file(filename, { superuser: "try" });
+
+    try {
+        await file.replace("");
+    } catch (err) {
+        console.warn("Cannot create new file", err);
+        throw err;
+    }
+
+    if (owner !== undefined) {
+        try {
+            await cockpit.spawn(["chown", owner, filename], { superuser: "require" });
+        } catch (err) {
+            console.warn("Cannot chown new file", err);
+            throw err;
+        }
+    }
+
+    if (content) {
+        // We need to obtain the tag to retain file ownership
+        // Can't do this async because we can't get the tag via await
+        await cockpit.file(filename, { superuser: "try" }).read()
+                .then(((async (_content: string, tag: string) => {
+                    try {
+                        await cockpit.file(filename, { superuser: "try" }).replace(content, tag);
+                    } catch (err) {
+                        console.warn("Cannot set initial file text", err);
+                        throw err;
+                    }
+                }) as any /* eslint-disable-line @typescript-eslint/no-explicit-any */));
+    }
+}
+
+const CreateFileModal = ({ dialogResult, path } : {
+    dialogResult: DialogResult<string>,
+    path: string
+}) => {
+    const [filename, setFilename] = React.useState<string>("");
+    const [owner, setOwner] = React.useState<string | undefined>();
```

Comment:
I'd advise to initialize this to `null`,  like filenameError and crateError (and most of other places in cockpit). `undefined` is usually for "I typoed the variable name" or something such.